### PR TITLE
feat: use dotenv

### PR DESCRIPTION
**Pull request by kulame**
dotenv support

### CODE REVIEW

File: .env.test [hunk 1]
```diff
@@ -0,0 +1 @@
+LEMMY_DATABASE_URL="postgresql://dbuser:dbuser@localhost:5432/lemmy"
```
Comment (dessalines) [inline .env.test:1]: Could you remove this file? Thx.

**Review by dessalines — CHANGES_REQUESTED** [review]

File: .env.test (hunk 1 above)
Comment (kulame) [reply .env.test]: Already deleted.

**Comment by Nutomic** [comment]
Whats the purpose of this change? You can already set the database url via env var or config file. Adding another way to set it would only cause confusion.

**Comment by kulame** [comment]
> 

It is difficult to modify the system environment variables in some execution environments, and using dotenv as a replacement is a convenient way

**Comment by Nutomic** [comment]
You should give more details what problem you are trying to solve like this, and why the config file is not sufficient. Keep in mind that we will have to support this functionality forever if it gets merged, so there needs to be a good reason.

**Comment by dessalines** [comment]
I checked the docs, its just that line. I'm not exactly sure why you'd need a `.env` file in addition to the already externalized `lemmy.hjson` file tho.

**Comment by dessalines** [comment]
We can probably close this, since the `lemmy.hjson` should be enough.

**Pull request closed without merging**